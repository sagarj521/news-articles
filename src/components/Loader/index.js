import React from 'react';
import './Loader.css';

const Loader = () => {
	return <div className="loading" data-testid="loader" role="presentation"></div>;
};

export default Loader;
