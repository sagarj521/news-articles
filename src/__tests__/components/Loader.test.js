import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Loader from "../../components/Loader";

test("renders without crashing", () => {
  const { container } = render(<Loader />);
  expect(container).toBeInTheDocument();
});

test("matches snapshot", () => {
  const { asFragment } = render(<Loader />);
  expect(asFragment()).toMatchSnapshot();
});
