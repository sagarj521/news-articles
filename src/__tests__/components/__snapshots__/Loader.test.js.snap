// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading"
  />
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading"
  />
</DocumentFragment>
`;
